CMakeLists.txt

```diff
@@ -1,16 +1,24 @@
 cmake_minimum_required(VERSION 3.5)
 project(LibreCuda CXX)
 
-# create elf-parser library
-add_subdirectory(third_party/ELFIO)
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+include_directories(
+        ${CMAKE_SOURCE_DIR}/src
+)
 
-add_executable(LibreCuda
-        src/main.cpp
-        src/librecuda.cpp
-        src/librecuda_status.cpp
-        src/cmdqueue.cpp)
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/third_party/ELFIO/elfio)
+        message("-- ELFIO not found, fetching ELFIO...")
+        execute_process(COMMAND git submodule update --init)
+else ()
+        message("-- ELFIO found.")
+        add_subdirectory(third_party/ELFIO)
+endif()
 
-target_link_libraries(LibreCuda PRIVATE elfio)
+# create elf-parser library
+add_subdirectory(src)
```

Comment:
src should not become a CMake module directory.
Consider
tests/
tests/CmakeLists.txt

(Note that this is technically currently a replacement for the so called cuda driver api., not the cuda runtime, etc. It should be named accordingly)
driverapi/CMakeLists.txt
driverapi/src/